### PR TITLE
error handling for use-package #246

### PR DESCRIPTION
**Pull request by edvorg**
Hi.
I suggest a way to handle errors from packages loading process by overriding use-package handler

### CODE REVIEW

File: use-package.el [hunk 1]
```diff
@@ -168,7 +168,7 @@ convert it to a string and return that."
 (defun use-package-load-name (name &optional noerror)
   "Return a form which will load or require NAME depending on
 whether it's a string or symbol."
-  (if (stringp name) `(load ,name 'noerror) `(require ',name nil 'noerror)))
```
Comment (npostavs) [inline use-package.el]: Oh, I made a mistake in #225, those `'noerror`s should have been `,noerror`. You should probably drop the `noerror` param if you're going to ignore it anyway.

Comment (edvorg) [inline use-package.el]: The error goes into `use-package-require-error-handler`. So we need to omit 'noerror param here.

Comment (npostavs) [inline use-package.el]: Yes, I'm saying you should therefore drop the `&optional noerror` part too.

Comment (edvorg) [inline use-package.el]: Oh. I understood. Thank you. I will fix it.

**Comment by jwiegley** [comment]
@edvorg This is not working right for me, I'm getting lots of messages like this after merging:

```
Error (use-package): helm-config :config: Wrong number of arguments: if, 1
Error (use-package): dired :config: Wrong number of arguments: if, 1
Error (use-package): bookmark :config: Wrong number of arguments: if, 1
Error (use-package): cus-edit :config: Wrong number of arguments: if, 1
Error (use-package): paredit :config: Wrong number of arguments: if, 1
Error (use-package): calfw :init: Wrong number of arguments: if, 1
```

And in `*Messages*`:

```
Eager macro-expansion failure: (wrong-number-of-arguments if 1)
File mode specification error: (wrong-number-of-arguments if 1)
```

Reverting until these can be tracked down, which I will not have time myself to do for a bit.

**Pull request merged**